File: straights.py
from sys import stdin,stdout

N = int(stdin.readline())
cards = sorted(map(int,stdin.readline().split())) #O(N log N)
frequency = {} #sum of values in hash table = minimum number of turns
for c in cards:
    frequency[c] = 1 if c not in frequency else frequency[c] + 1
    if c - 1 in frequency: #part of a consecutive sequence
        frequency[c-1] = max(0,frequency[c-1]-1) #only subtract 1 if frequency[c-1] > 0
stdout.write(f'{sum(frequency.values())}')
    

File: straights.java
import java.util.*;
import java.io.*;

public class straights {
    public static void main(String args[]) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        PrintWriter pw = new PrintWriter(new BufferedWriter(new OutputStreamWriter(System.out)));
        br.readLine(); //value of N not required
        int[] sequence = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
        Arrays.sort(sequence); // O(N log N)
        HashMap<Integer,Integer> frequencyTable = new HashMap<>();
        for (int c: sequence){
            frequencyTable.merge(c, 1, (a,b) -> a + b);
            if (frequencyTable.containsKey(c-1)){
                frequencyTable.put(c-1,Math.max(0,frequencyTable.get(c-1)-1)); //only subtract 1 if frequency[c-1] > 0
            }
        }
        int minimum = 0;
        for (int value: frequencyTable.values()) minimum += value;
        pw.println(minimum);
        br.close();
        pw.close();
    }
}